content/articles/searching-and-sorting-in-matlab/index.md

```diff
@@ -0,0 +1,89 @@
+search. In this method, the target number that is being sought is compared
+with the first member of the database, then with the second, etc., until either
+the number is found, or the end of the database is reached, whichever comes
+first. Instead of dealing with a real database, let’s work on the simplest version of the search problem. We are given a vector of numbers and a target
+value. Our task is to search for that target value in the vector. If we find it,
+then the answer we give is the index of the element that we found.
+If we do not find it, we return an impossible index, say, −1, as a flag that indicates that the search failed. That flag is crucial to the search algorithm because there must be some means of informing the searcher that the target is
+not there
+“flag” means a value indicating a special condition. In this case, the condition
+is “target not found”, or “search failure”.)
+The function, sequential_search, below carries out the sequential search:
+```matlab
+function index = sequential_search ...
+(vector,target,first,last)
+%SEQUENTIAL_SEARCH
+% SEQUENTIAL_SEARCH(VECTOR,TARGET,FIRST,LAST) returns
+% smallest index for which TARGET == VECTOR(index) or
+% -1, if TARGET not found within VECTOR(FIRST:LAST).
+found = false; % Assume the target is not in vector
+for n = first:last
+ if target == vector(n)
+ found = true; % We found it...
+ break; % so we quit looking for it!
+ end
+end
+index = n;
+if ~found
+ index = -1;
+end
+```
+First we note that we have used the line continuation operator (...) to continue a statement on to the next line. Such continuation has nothing to do
+with functionality. Next we note that that there are four arguments. Argument 1 is the vector to be searched; Argument 2 is the target that we are
+searching for; Arguments 3 and 4 give the beginning and the end, respectively, of the range of indices over which we wish to search. Most of the time,
+when we call the function, we will set first to 1 and last to the full length
+of vector, but sometimes we may wish to search only a limited range. Arguments 3 and 4 are there to provide this option. So for example, if we wish to
+search all of a vector, named accounts, for the number stored in smith, we
+would give the call,
+```matlab
+sequential_search(accounts, smith, 1, length(accounts))
+```
+but, if we wish to search only within, say, accounts(450:5600), we would
+give the call,
+```matlab
+sequential_search(accounts, smith, 450, 5600)
+```
+As promised, this function returns –1, as a flag indicating failure, where failure means that the target value is not equal to any of the elements in the vector. To assist in setting the flag, we have used an internal flag called found
+and the built-in functions false and true
+
+### Sorting
+A sorting algorithm that is not very efficient, but is simple enough to be readily understood is the selection sort. Here is a function that implements the
+selection-sort algorithm:
```

Comment:
Do simple explanations on your matlab code using comments to help readers understand the code more